src/RA_Achievement.cpp

```diff
@@ -54,149 +30,112 @@ void Achievement::Parse( const Value& element )
 
 	if( element["MemAddr"].IsString() )
 	{
-		std::string sMem = element["MemAddr"].GetString();
-		char buffer[8192];
-		sprintf_s( buffer, 8192, sMem.c_str() );
-		ParseMemString( buffer );
+		const char* sMem = element["MemAddr"].GetString();
+		if (!m_vConditions.ParseFromString(sMem) || *sMem != '\0') {
+			ASSERT(!"Invalid MemAddr");
+			m_vConditions.Clear();
+		}
```

Comment:
Similar to RetroArch, this will disable an achievement if there's an issue parsing it. `ConditionSet.Test` will return false if there aren't any groups.